src/http/errors.ts

```diff
@@ -1,23 +1,43 @@
-import makeError from 'make-error'
+import {BaseError} from 'make-error'
```

Comment:
Does it still make sense to use `make-error`? I think we reached for it before ES6 - could we just go with `extends Error` these days?